boxcli/add.go

```diff
@@ -27,6 +30,31 @@ func addCmdFunc() runFunc {
 		if err != nil {
 			return errors.WithStack(err)
 		}
-		return box.Add(args...)
+
+		if err = box.Add(args...); err != nil {
```

Comment:
I think you have too much logic in here for `add`, `rm` and `shell`.

The goal of the `devbox` _library_ is that it can be used as an SDK that is _equivalent_ to using the CLI. In other words, `box.Add("package")` should be roughly the same as calling `devbox add package`; similarly `box.Remove("package")` should be roughly the same as calling `devbox rm package`. But since you've added all this logic inside of the CLI, instead of inside of the library, that is no longer true.